mantidproject/mantid
# Update plot creation args workspace name during workspace rename

**Description of work.**

Previously, if a user made a plot and renamed the workspace, they would not be able to change the normalisation without replotting. This fixes that by updating the workspace name in the creation arguments of the axis, as this is what is used for the workspace name check during the rename event and was not previously updated.

**To test:**

1. Plot any data
2. Check you're able to change the normalisation
3. Rename the workspace in the ADS
4. Confirm you're still able to change the normalisation on the plot without creating a new one

Fixes #30739 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
